### PR TITLE
Adding supplier order payment types and vcc cancellation and modifica…

### PR DESCRIPTION
**Pull request by mingazhev**
…tion support

### CODE REVIEW

File: Api/Services/Accommodations/Bookings/BookingExecution/BookingRegistrationService.cs [hunk 1]
```diff
@@ -157,7 +160,8 @@ Task CreateSupplierOrder(Booking booking)
                 HtId = htId,
                 Tags = tags,
                 IsDirectContract = isDirectContract,
-                CancellationPolicies = availabilityInfo.RoomContractSet.Deadline.Policies
+                CancellationPolicies = availabilityInfo.RoomContractSet.Deadline.Policies,
+
```
Comment (oleg1std) [inline Api/Services/Accommodations/Bookings/BookingExecution/BookingRegistrationService.cs:164]: extra space

**Review by oleg1std — APPROVED** [review]

**Review by AlexeyShok — APPROVED** [review]

File: HappyTravel.Edo.Common/Enums/SupplierPaymentType.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace HappyTravel.Edo.Common.Enums
+{
+    public enum SupplierPaymentType
+    {
+        None = 0,
+        DirectPayments = 1,
+        CreditCard = 2
```
Comment (kirillta) [inline HappyTravel.Edo.Common/Enums/SupplierPaymentType.cs:7]: SupplierPaymentType**s** and DirectPayment or CreditCard**s**, I guess

File: HappyTravel.Edo.CreditCards/Services/VirtualCreditCardProvider.cs [hunk 3]
```diff
@@ -27,8 +27,17 @@ public async Task<Result<CreditCardInfo>> Get(string referenceCode, MoneyAmount
                 HolderName: virtualCreditCard.Holder,
                 SecurityCode: virtualCreditCard.Code);
         }
-        
-        
+
+
+        public async Task<Result> ProcessAmountChange(string referenceCode, MoneyAmount newAmount)
+        {
+            if (newAmount.Amount == 0)
```
Comment (kirillta) [inline HappyTravel.Edo.CreditCards/Services/VirtualCreditCardProvider.cs:34]: Is it Ok to compare different types?

File: Api/Services/SupplierOrders/SupplierOrderService.cs [hunk 4]
```diff
@@ -14,15 +15,16 @@ namespace HappyTravel.Edo.Api.Services.SupplierOrders
 {
     public class SupplierOrderService : ISupplierOrderService
```
Comment (kirillta) [inline Api/Services/SupplierOrders/SupplierOrderService.cs:16]: `SupplierOrder` sounds not clear in general. Maybe we should rename it.

**Pull request merged**